### PR TITLE
Add additional rendering statistics

### PR DESCRIPTION
**Pull request by reichlfl**
On top of the required memory, this adds several additional statistics to the batchers that will be listed in the debug tab.

This also changes the statistics reporting to return a struct collecting all statistics, so it's easier to extend in the future.

### CODE REVIEW

File: src/OrbitGl/Batcher.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2023 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitGl/Batcher.h"
+
+namespace orbit_gl {
+bool operator==(const Batcher::Statistics& lhs, const Batcher::Statistics& rhs) {
```
Comment (florian-kuebler) [inline src/OrbitGl/Batcher.cpp:8]: Why don't you implement that in the struct/class itself?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

**Review by florian-kuebler — APPROVED** [review]
I guess, lgtm.

I am in general not sure about the "devmode" only code, but I guess we can live with that.

File: src/OrbitGl/include/OrbitGl/Batcher.h [hunk 2]
```diff
@@ -32,6 +39,8 @@ class Batcher : public BatcherInterface {
   BatcherId batcher_id_;
 };
 
+[[nodiscard]] bool operator==(const Batcher::Statistics& lhs, const Batcher::Statistics& rhs);
```
[the remark this answers is not in the document]
Comment (reichlfl) [reply src/OrbitGl/include/OrbitGl/Batcher.h]: Done, thanks for pointing it out.

File: src/OrbitGl/Batcher.cpp (hunk 1 above)
Comment (reichlfl) [reply src/OrbitGl/Batcher.cpp]: Hm I thought we had the rule to declare unary operators within the class, and binary / tertiary outside in the style guide, but I was wrong. Anyways that is usually how I treat it, it's more consistent if the arguments can be cast, or are of different types and you'd want to provide both orders of parameters.

**Pull request merged**